src/components/app/details/appDetails/DeploymentStatusBreakdown.tsx

```diff
@@ -1,79 +1,59 @@
+import { useRouteMatch } from 'react-router-dom'
 import ErrorBar from '../../../common/error/ErrorBar'
 import IndexStore from '../../../v2/appDetails/index.store'
-import { useRouteMatch } from 'react-router-dom'
+import { RenderStatusDetailRow } from './RenderStatusDetailRow'
 
 export default function DeploymentStatusDetailBreakdown({
     deploymentStatusDetailsBreakdownData,
+    streamData,
 }: DeploymentStatusDetailBreakdownType) {
     const _appDetails = IndexStore.getAppDetails()
     const { url } = useRouteMatch()
+    
+    return (<>
+        {!url.includes(`/${URLS.APP_CD_DETAILS}`) && <ErrorBar appDetails={_appDetails} />}
+        <div className="deployment-status-breakdown-container pl-20 pr-20 pt-20 pb-20">
+            <RenderStatusDetailRow
+                type={TIMELINE_STATUS.DEPLOYMENT_INITIATED}
+                deploymentDetailedData={deploymentStatusDetailsBreakdownData}
+            />
+            <ErrorInfoStatusBar
+                type={TIMELINE_STATUS.GIT_COMMIT}
+                nonDeploymentError={deploymentStatusDetailsBreakdownData.nonDeploymentError}
+                errorMessage={deploymentStatusDetailsBreakdownData.deploymentError}
+            />
+            <RenderStatusDetailRow type={TIMELINE_STATUS.GIT_COMMIT} deploymentDetailedData={deploymentStatusDetailsBreakdownData} />
+            <ErrorInfoStatusBar
+                type={TIMELINE_STATUS.KUBECTL_APPLY}
+                nonDeploymentError={deploymentStatusDetailsBreakdownData.nonDeploymentError}
+                errorMessage={deploymentStatusDetailsBreakdownData.deploymentError}
+            />
+            <RenderStatusDetailRow type={TIMELINE_STATUS.KUBECTL_APPLY}deploymentDetailedData={deploymentStatusDetailsBreakdownData} />
+            <RenderStatusDetailRow
+                type={TIMELINE_STATUS.APP_HEALTH}
+                hideVerticalConnector={true}
+                deploymentDetailedData={deploymentStatusDetailsBreakdownData}
+                streamData={streamData}
+            />
+        </div></>
+    )
+}
 
-    const renderIcon = (iconState: string): JSX.Element => {
-        switch (iconState) {
-            case 'success':
-                return <Check className="icon-dim-20 green-tick" />
-            case 'failed':
-                return <Error className="icon-dim-20" />
-            case 'unknown':
-                return <Question className="icon-dim-20" />
-            case 'inprogress':
-                return <div className="pulse-highlight"></div>
-            case 'unreachable':
-                return <Close className="icon-dim-20" />
-            default:
-                return <Timer className="icon-dim-20 timer-icon" />
-        }
-    }
-    const renderStatusDetailRow = (key: string, hideVerticalConnector?: boolean): JSX.Element => {
-        return (
+function ErrorInfoStatusBar({ nonDeploymentError, type, errorMessage }) {
```

Comment:
Try to keep Only one function in a single tsx component. Please do Extract out this function

Reply:
Type is missing